src/util/helpers/helpers.cpp

```diff
@@ -437,3 +439,40 @@ std::string GenerateRandomString(const size_t length, const std::string_view cha
 
 	return result;
 }
+
+std::optional<std::vector<uint8>> zlibDecompress(const std::vector<uint8>& compressed)
+{
+	int err;
+	std::vector<uint8> decompressed;
+	size_t outWritten = 0;
+	z_stream stream;
+	stream.zalloc = Z_NULL;
+	stream.zfree = Z_NULL;
+	stream.opaque = Z_NULL;
+	stream.avail_in = compressed.size();
+	stream.next_in = (Bytef*)compressed.data();
+	err = inflateInit2(&stream, 32); // 32 is a zlib magic value to enable header detection
+	if (err != Z_OK)
+		return {};
+
+	do
+	{
+		decompressed.resize(decompressed.size() + 1024);
```

Comment:
Increasing by a fixed 1024 bytes at a time seems rather inefficient? For a 1MB file thats a thousand vector resize operations.
I recommend starting out with a smallish size like 32KB (maybe even as a size hint parameter?) and then just doubling the size each loop iteration.

Reply:
See latest commit. I know defining a variable and assigning the argument is redundant but I think it helps readability and it should be trivial for a compiler to optimize. I set the hint in the game list for the icons to 70KiB since one extracted game icon TGA I looked at is slightly more than 64KiB.